### ISSUE TITLE
no-for-loop should detect and ignore variables populated with DOM node if possible

### ISSUE THREAD

**Issue by Slapbox** [sindresorhus/eslint-plugin-unicorn#1531]
For loops for non-iterable DOM nodes are treated as iterable arrays.

**Affected rule:** no-for-loop

**Input:**
```js
const visibleItems = document.querySelector('.visible');
for (let x = 0; x < visibleItems.length; x++) {
  someFunc(visibleItems[x])
}
```

**Output:**

```js
const visibleItems = document.querySelector('.visible');
for (const [x, visibleItem] of visibleItems.entries()) {
  someFunc(visibleItem)
}
```

-----

As an aside, the fact that the auto-fix depluralizes `visibleItems[x]` to `visibleItem` is a very nice touch!

**Comment by sindresorhus** [comment]
Your code is invalid. `querySelector` is not iterable because it returns a single element. I think you meant to use `querySelectorAll` which is actually iterable.

**Comment by fregante** [comment]
Some notes are not iterable but do have an index signature 😬

https://www.w3schools.com/jsref/coll_form_elements.asp

However they're likely to be old APIs that have a better alternative nowadays, like form.fields.length

**Comment by Slapbox** [comment]
@sindresorhus truth, sloppy demo code on my part.

Our actual code uses `getElementsByClassName('visible')` because it's pretty heavy to - even though this is probably silly ([and quite possibly misguided](https://dev.to/wlytle/performance-tradeoffs-of-queryselector-and-queryselectorall-1074))

Lesson? Don't try to "cleanup" demo code because there's often something you don't know.

As @fregante points out - indeed, an older API, but it does break valid code (even if the demo was not valid.) 

**So let's try this again, this time without the oops.**

**Input:**
```js
const visibleItems = document.getElementsByClassName('visible');
for (let x = 0; x < visibleItems.length; x++) {
  someFunc(visibleItems[x])
}
```

**Output:**
```js
const visibleItems = document.getElementsByClassName('visible');
for (const [x, visibleItem] of visibleItems.entries()) {
  someFunc(visibleItem)
}
```

**Comment by fregante** [comment]
In short, `no-for-loop` breaks on array-like objects.
